### PR TITLE
Integration tests for the backend response

### PR DESCRIPTION
**Pull request by cailyncodes**
In progress towards #940

### CODE REVIEW

File: src/Hedwig/Controllers/TestController.cs [hunk 1]
```diff
@@ -0,0 +1,23 @@
+using Microsoft.AspNetCore.Mvc;
```
Comment (cailyncodes) [inline src/Hedwig/Controllers/TestController.cs:1]: Oops need to delete

File: test/HedwigTests/Fixtures/TestApiProvider.cs [hunk 2]
```diff
@@ -1,40 +1,74 @@
...
+				.UseStartup<TestStartup>()
+				.UseTestServer();
+		}
+	}
+}
 
-			var builder = new WebHostBuilder()
-				.UseConfiguration(config)
-				.UseStartup<TestStartup>();
+// using System;
```
Comment (cailyncodes) [inline test/HedwigTests/Fixtures/TestApiProvider.cs:26]: Delete

File: test/HedwigTests/Fixtures/TestStartup.cs [hunk 3]
```diff
@@ -15,8 +14,6 @@ public override void ConfigureServices(IServiceCollection services)
 			base.ConfigureServices(services);
 			services.ConfigureSqlServer(Configuration.GetConnectionString("HEDWIG"));
 
-			IHttpContextAccessor httpContextAccessor = new TestHttpContextAccessorProvider().HttpContextAccessor;
```
Comment (cailyncodes) [inline test/HedwigTests/Fixtures/TestStartup.cs:18]: This seemed to be unnecessary for any current tests and broke integration tests.

File: test/HedwigTests/HedwigTests.csproj [hunk 4]
```diff
@@ -7,6 +7,7 @@
 
   <ItemGroup>
     <PackageReference Include="Microsoft.AspNetCore.TestHost" Version="3.0.0" />
+		<PackageReference Include="Microsoft.AspNetCore.Mvc.Testing" Version="3.0.0" />
```
Comment (cailyncodes) [inline test/HedwigTests/HedwigTests.csproj:10]: Needed for the new testserver/testclient approach

File: test/HedwigTests/Integrations/EnrollmentsControllerIntegrationTests.cs [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+			Assert.Equal(enrollment.Id, responseEnrollment.Id);
+			Assert.Equal(enrollment.ChildId, responseEnrollment.ChildId);
+			Assert.NotNull(responseEnrollment.Child);
+			Assert.Equal(enrollment.ChildId, responseEnrollment.Child.Id);
+			Assert.Equal(enrollment.Child.FirstName, responseEnrollment.Child.FirstName);
+			Assert.Equal(enrollment.Child.MiddleName, responseEnrollment.Child.MiddleName);
+			Assert.Equal(enrollment.Child.LastName, responseEnrollment.Child.LastName);
+			Assert.Empty(responseEnrollment.Child.Enrollments);
+			Assert.Equal(enrollment.SiteId, responseEnrollment.SiteId);
+			// responseEnrollment.Site is null here and I don't know why?
```
Comment (cailyncodes) [inline test/HedwigTests/Integrations/EnrollmentsControllerIntegrationTests.cs:75]: Any help/ideas?

File: test/HedwigTests/Integrations/EnrollmentsControllerIntegrationTests.cs [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+			// It is not null in the response on the client
+			// Assert.NotNull(responseEnrollment.Site.Id);
+			// Assert.NotNull(responseEnrollment.Site.Name);
+			Assert.Equal(enrollment.AgeGroup, responseEnrollment.AgeGroup);
+			Assert.Equal(enrollment.Entry, responseEnrollment.Entry);
+			Assert.Equal(enrollment.Exit, responseEnrollment.Exit);
+			Assert.Equal(enrollment.ExitReason, responseEnrollment.ExitReason);
+			// Fundings
```
Comment (cailyncodes) [inline test/HedwigTests/Integrations/EnrollmentsControllerIntegrationTests.cs:83]: Need to figure out the best way to assert for collection. 1) Don't worry about matching the data and just assert.all and assert.notnull within the all statement. 2) Sort and iterate and match. 3) Assert for contains of collections of each property

File: test/HedwigTests/Integrations/HedwigAPIRequests.cs [hunk 7]
```diff
@@ -0,0 +1,47 @@
+using System;
+using System.Net.Http;
+using Hedwig.Models;
+using Microsoft.AspNetCore.WebUtilities;
+
+namespace HedwigTests.Integrations
+{
+	public static class HedwigAPIRequests
```
Comment (cailyncodes) [inline test/HedwigTests/Integrations/HedwigAPIRequests.cs:8]: This is way I was envisioning that we would add the collection of all API requests that ECE Reporter uses

File: test/HedwigTests/xunit.runner.json [hunk 8]
```diff
@@ -0,0 +1,3 @@
+{
+  "shadowCopy": false
```
Comment (cailyncodes) [inline test/HedwigTests/xunit.runner.json:2]: Not sure this is actually needed. Some tutorial told me to add it but I think I ventured far away from that by the time I got it to work.

File: test/HedwigTests/Integrations/EnrollmentsControllerIntegrationTests.cs (hunk 5 above)
Comment (cailyncodes) [reply test/HedwigTests/Integrations/EnrollmentsControllerIntegrationTests.cs:75]: The response JSON has site fields, but it gets lost in deseralization :/

File: test/HedwigTests/Integrations/EnrollmentsControllerIntegrationTests.cs (hunk 6 above)
Comment (cailyncodes) [reply test/HedwigTests/Integrations/EnrollmentsControllerIntegrationTests.cs:83]: Went with 1

File: test/HedwigTests/xunit.runner.json (hunk 8 above)
Comment (cailyncodes) [reply test/HedwigTests/xunit.runner.json:2]: removed it and it seems to work (hoping this fixes the error in the pipeline)

File: test/HedwigTests/Integrations/EnrollmentsControllerIntegrationTests.cs (hunk 5 above)
Comment (cailyncodes) [reply test/HedwigTests/Integrations/EnrollmentsControllerIntegrationTests.cs:75]: I think it's because of protected set

Comment (cailyncodes) [reply test/HedwigTests/Integrations/EnrollmentsControllerIntegrationTests.cs:75]: Resolved!

File: test/HedwigTests/Integrations/EnrollmentsControllerIntegrationTests.cs [hunk 9]
```diff
@@ -0,0 +1,155 @@
...
+			var response = await client.SendAsync(request);
+			response.EnsureSuccessStatusCode();
+
+			var responseString = await response.Content.ReadAsStringAsync();
+			var responseEnrollments = JsonConvert.DeserializeObject<List<Enrollment>>(responseString);
+			Assert.NotEmpty(responseEnrollments);
+
+			var responseEnrollment = responseEnrollments.First();
+
+			Assert.Equal(enrollment.Id, responseEnrollment.Id);
```
Comment (jlhogan) [inline test/HedwigTests/Integrations/EnrollmentsControllerIntegrationTests.cs:59]: :raised_hands: a sigh of relief

File: test/HedwigTests/xunit.runner.json (hunk 8 above)
Comment (cailyncodes) [reply test/HedwigTests/xunit.runner.json]: Well it wasn't needed but didn't help pipeline errors. Notwithstanding, I think I finally got the pipelines to pass!

**Pull request merged**